### PR TITLE
Fix data table ph1

### PR DESCRIPTION
**Pull request by biaoli0**
### Changes:
Suggest to review commit by commit.

### CODE REVIEW

**Comment by IgorNadj** [comment]
Hi, which cards do these relate to?

**Comment by biaoli0** [comment]
@IgorNadj Any issues found by Andrew in the epic branch.

**Comment by biaoli0** [comment]
1. Add power to edit fields, this will introduce a runtimeParameter state ([Issue 3](https://linear.app/bes/issue/RN-731#comment-743a42e9) [Issue 4](https://linear.app/bes/issue/RN-731#comment-29a32661) [Issue 5](https://linear.app/bes/issue/RN-731#comment-12d7fede) [Issue 6](https://linear.app/bes/issue/RN-731#comment-12d7fede) from RN-731,[ Issue 3](https://linear.app/bes/issue/RN-733#comment-8bcaa4cb) from RN-733)   
2. Fix showing error in sql editor if more than one custom parameter, tiny ([Issue 3](https://linear.app/bes/issue/RN-729#comment-c3e20c82) in RN-729)
3. Remove condition check in preview filter ([Issue 2](https://linear.app/bes/issue/RN-733#comment-28ce210f) [Issue 5](https://linear.app/bes/issue/RN-733#comment-5d97ac9a) from RN-733)

File: packages/admin-panel/src/dataTables/useRuntimeParameters.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Tupaia
+ * Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import { useCallback } from 'react';
```
Comment (IgorNadj) [inline packages/admin-panel/src/dataTables/useRuntimeParameters.js:6]: Note: this is a new hook to separate runtime params state out from the rest of the state.

File: packages/admin-panel/src/dataTables/components/PreviewFilters/filters/BooleanField.js [hunk 2]
```diff
@@ -4,15 +4,15 @@
...
 import PropTypes from 'prop-types';
-import React, { useEffect } from 'react';
+import React from 'react';
 
 import { Select } from '@tupaia/ui-components';
 import { ParameterType } from '../../editing';
 
 export const BooleanField = ({ name, value, onChange }) => {
-  useEffect(() => {
+  if (!value) {
```
Comment (IgorNadj) [inline packages/admin-panel/src/dataTables/components/PreviewFilters/filters/BooleanField.js:13]: Looks like the issue that this solves is actually solved by default props changing undefined to null

**Review by IgorNadj — APPROVED** [review]
LGTM

File: packages/admin-panel/src/dataTables/components/PreviewFilters/filters/BooleanField.js [hunk 3]
```diff
@@ -9,20 +9,20 @@ import React from 'react';
 import { Select } from '@tupaia/ui-components';
 import { ParameterType } from '../../editing';
 
-export const BooleanField = ({ name, value, onChange }) => {
-  if (!value) {
-    onChange(false);
-  }
+export const BooleanField = ({ name, value, onChange, config }) => {
+  const defaultValue = config?.hasDefaultValue ? config?.defaultValue.toString() : null;
```
Comment (IgorNadj) [inline packages/admin-panel/src/dataTables/components/PreviewFilters/filters/BooleanField.js:13]: I got an error here "Uncaught TypeError: Cannot read properties of undefined (reading 'toString')"

**Pull request merged**